packages/block-editor/src/components/embedded-admin-context/index.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { useRefEffect } from '@wordpress/compose';
+import { useState, createPortal } from '@wordpress/element';
+import { ENTER, SPACE, ESCAPE } from '@wordpress/keycodes';
+import { focus } from '@wordpress/dom';
+import { __experimentalStyleProvider as StyleProvider } from '@wordpress/components';
+
+export default function EmbeddedAdminContext( props ) {
+	const [ shadow, setShadow ] = useState();
+	const [ hasFocus, setHasFocus ] = useState();
+	const ref = useRefEffect( ( element ) => {
+		const root = element.attachShadow( { mode: 'open' } );
+		const style = document.createElement( 'style' );
+		Array.from( document.styleSheets ).forEach( ( styleSheet ) => {
+			if ( styleSheet.ownerNode.getAttribute( 'data-emotion' ) ) {
```

Comment:
Feels weird to target `data-emotion`. That's sounds like an implementation detail that shouldn't be relied upon.

Reply:
Yeah, ideally we should only include the styles that are needed for the placeholder, but that's very hard to do. Not sure what else we can do here, aside from just including everything. It's not really a problem to include the styles, it's just bad for performance.